pkg/monitor/api.go

```diff
@@ -182,6 +193,117 @@ func startServerMonitoring(ctx context.Context, m *Monitor, clusterConfig *rest.
 	return nil
 }
 
+func startNetworkMonitoring(ctx context.Context, m *Monitor, clusterConfig *rest.Config, timeout time.Duration, resourceLocator, url string, disableConnectionReuse bool) error {
+	kubeTransportConfig, err := clusterConfig.TransportConfig()
+	if err != nil {
+		return err
+	}
+	tlsConfig, err := transport.TLSConfigFor(kubeTransportConfig)
+	if err != nil {
+		return err
+	}
+	var httpTransport *http.Transport
+
+	if disableConnectionReuse {
+		httpTransport = &http.Transport{
+			Dial: (&net.Dialer{
+				Timeout:   timeout,
+				KeepAlive: -1, // this looks unnecessary to me, but it was set in other code.
+			}).Dial,
+			TLSClientConfig:     tlsConfig,
+			TLSHandshakeTimeout: timeout,
+			DisableKeepAlives:   true, // this prevents connections from being reused
+			IdleConnTimeout:     timeout,
+		}
+	} else {
+		httpTransport = &http.Transport{
+			Dial: (&net.Dialer{
+				Timeout: timeout,
+			}).Dial,
+			TLSClientConfig:     tlsConfig,
+			TLSHandshakeTimeout: timeout,
+			IdleConnTimeout:     timeout,
+		}
+	}
+
+	roundTripper := http.RoundTripper(httpTransport)
+	if kubeTransportConfig.HasTokenAuth() {
+		roundTripper, err = transport.NewBearerAuthWithRefreshRoundTripper(kubeTransportConfig.BearerToken, kubeTransportConfig.BearerTokenFile, httpTransport)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	httpClient := http.Client{
+		Transport: roundTripper,
+	}
+
+	go NewSampler(m, time.Second, func(previous bool) (condition *monitorapi.Condition, next bool) {
+		resp, err := httpClient.Get(clusterConfig.Host + url)
+
+		if err == nil && resp != nil && resp.Body != nil {
+			defer resp.Body.Close()
+		}
+
+		switch {
+		case err == nil && !previous:
+			condition = &monitorapi.Condition{
+				Level:   monitorapi.Info,
+				Locator: resourceLocator,
+				Message: fmt.Sprintf("%s started responding to network requests", resourceLocator),
+			}
+		case err != nil && IsNetworkError(err) && previous:
+			condition = &monitorapi.Condition{
+				Level:   monitorapi.Error,
+				Locator: resourceLocator,
+				Message: fmt.Sprintf("%s started failing: %v", resourceLocator, err),
+			}
+		}
+		return condition, IsNetworkError(err)
+	}).WhenFailing(ctx, &monitorapi.Condition{
+		Level:   monitorapi.Error,
+		Locator: resourceLocator,
+		Message: fmt.Sprintf("%s is not responding to network requests", resourceLocator),
+	})
+
+	return nil
+}
+
+var apiMachineryErrors = []string{
+	"connection refused", // TODO: Likely a networking issue, but require more analysis to tell
```

Comment:
we can document it, usually it happens something like this:
- we send a SYN
- we receive a RST

This means that there is actually some sort of communication because we receive a RST packet, who sends the RST is the key, it can be a firewall, kube-proxy does this for services without endpoints too, it can be the there is nothing listening on the host and the OS RSTs the connection to inform the client